collar_app/views.py

```diff
@@ -9,40 +9,68 @@
 def index(request):
 	return redirect('/dash')
 
-
-
+def market(request):
+	"""market any logged in user sees"""
+	try: #check if customer is logged in. This is in a try/except clause because if the user is not logged in the app will crash. The except clause will bring the user back to the login page.
+		context = {
+			"logged_user": User.objects.get(id=request.session['userid']),
+			"jobs": Job.objects.all()
+			}
+		return render(request, 'market.html', context)
+	except:
+		return redirect('/signin/login')
 
 def dash(request):
-	"""dashboard logged in user sees"""
+	"""dashboard: only logged in worker user sees"""
 	try: #check if customer is logged in. This is in a try/except clause because if the user is not logged in the app will crash. The except clause will bring the user back to the login page.
 		context = {
 			"logged_user": User.objects.get(id=request.session['userid']),
 			"jobs": Job.objects.all()
 			}
-		return render(request, 'dashboard.html', context)
+		if request.session['isworker']: # this page only for worker users
+			return render(request, 'dashboard.html', context)
+		else:
+			return render(request, 'marketplace.html', context)
 	except:
 		return redirect('/signin/login')
 
-
 def create(request):
 	"""workers can view jobs
 	brings in related_name """
-	try: # check ig user logged in
+	try: # check if user logged in
 		context = {
 			'logged_user': User.objects.get(id=request.session['user_id']), # if user is not logged in then this will jump to except clause
 			'users': User.objects.all(),
 			'jobs': Job.objects.all(),
 		}
-		return render(request, 'dashboard.html', context)
+		if request.session['isworker']: # this page only for worker users
+			return render(request, 'create.html', context)
+		else:
+			return render(request, 'marketplace.html', context)
 	except:
 		return redirect('/signin/login') # send to login page if not loggedin
 	
+def edit(request):
+	try: # check if user logged in
+		context = {
+			'logged_user': User.objects.get(id=request.session['user_id']), # if user is not logged in then this will jump to except clause
+			'users': User.objects.all(),
+			'jobs': Job.objects.all(),
+		}
+		if request.session['isworker']: # this page only for worker users
+			return render(request, 'edit.html', context)
+		else:
+			return render(request, 'marketplace.html', context)
+	except:
+		return redirect('/signin/login') # send to login page if not loggedin
 
+
+# redirects to login_reg_app to logout:
 def logout(request):
 	"""logout session, back to home page"""
 	request.session.flush()
 	print(request.session)
-	return redirect('/')
+	return redirect('/signin/logout')
```

Comment:
so this what was not keeping state correctly in the login flow?

Reply:
this will redirect to the login_reg_app to do the log out. It basically allows two different urls to logout: localhost/signin/logout and localhost/logout will both log the user out. It's working on my end

Reply:
Ok. will merge